From 6a31f1e08ff711ecbf2da41d12280004a4184d2b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Sat, 4 Aug 2018 14:24:35 +0200
Subject: [PATCH] Drop struct keyword

I am not entirely sure what's the problem here, probably C++11 vs C++14. I
think that in C++ you do not need the struct keyword here (as opposed to C at
some point) and it actually causes a compile error with c++11.
---
 src/sage/libs/eclib/wrap.cpp | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/sage/libs/eclib/wrap.cpp b/src/sage/libs/eclib/wrap.cpp
index 5fd5693b532..84da3d7f9ae 100644
--- a/sage/libs/eclib/wrap.cpp
+++ b/sage/libs/eclib/wrap.cpp
@@ -73,17 +73,17 @@ char* Curvedata_repr(struct Curvedata* curve)
   return stringstream_to_char(instore);
 }
 
-double Curvedata_silverman_bound(const struct Curvedata* curve)
+double Curvedata_silverman_bound(const Curvedata* curve)
 {
   return silverman_bound(*curve);
 }
 
-double Curvedata_cps_bound(const struct Curvedata* curve)
+double Curvedata_cps_bound(const Curvedata* curve)
 {
   return cps_bound(*curve);
 }
 
-double Curvedata_height_constant(const struct Curvedata* curve)
+double Curvedata_height_constant(const Curvedata* curve)
 {
   return height_constant(*curve);
 }
@@ -133,8 +133,8 @@ char* Curvedata_isogeny_class(struct Curvedata* E, int verbose)
 
 
 int mw_process(struct Curvedata* curve, struct mw* m,
-                      const struct bigint* x, const struct bigint* y,
-                      const struct bigint* z, int sat)
+                      const bigint* x, const bigint* y,
+                      const bigint* z, int sat)
 {
   Point P(*curve, *x, *y, *z);
   if (!P.isvalid())
@@ -188,7 +188,7 @@ int mw_rank(struct mw* m)
 }
 
 /* Returns index and unsat long array, which user must deallocate */
-int mw_saturate(struct mw* m, struct bigint* index, char** unsat,
+int mw_saturate(struct mw* m, bigint* index, char** unsat,
                        long sat_bd, int odd_primes_only)
 {
   vector<long> v;
@@ -238,12 +238,12 @@ char* two_descent_get_basis(struct two_descent* t)
   return p2point_vector_to_str(t->getbasis());
 }
 
-int two_descent_ok(const struct two_descent* t)
+int two_descent_ok(const two_descent* t)
 {
   return t->ok();
 }
 
-long two_descent_get_certain(const struct two_descent* t)
+long two_descent_get_certain(const two_descent* t)
 {
   return t->getcertain();
 }
